stats-test.cpp

```diff
@@ -2,10 +2,11 @@
 
 #include "catch.hpp"
 #include "stats.h"
+#include<math.h>
 
 #include <cmath>
 
-TEST_CASE("reports average, minimum and maximum") {
+TEST_CASE("reports average, minimum and maximum of array with all non-NAN elements") {
```

Comment:
Precise description of tests 👍

Reply:
I have updated the code with a new function which is removing nan elements from the given vector. Then I am applying the STL built in functions. This is because, by directly using the STL functions the output was incorrect for a few tests such as only vector having only 1 nan element. I also added a new test case for array with all nan elements.

Reply:
Excellent decision to return a new list after removing NaNs 👍